unattended-upgrade

```diff
@@ -453,6 +454,38 @@ class Unlocked:
             apt_pkg.pkgsystem_lock()
 
 
+class OriginPkgs(defaultdict):
+    """
+    Package names by highest allowed pretty-printed origin
+
+    """
+    def add(self, pkg, cache):
+        # type: (apt.Package, UnattendedUpgradesCache) -> None
+        allowed_origins = cache.allowed_origins
+        for origin in ver_in_allowed_origin(pkg, allowed_origins).origins:
+            if is_allowed_origin(origin, allowed_origins):
+                self[origin.origin + " " + origin.archive].add(pkg.name)
+                return
+
+    def pop_upgradable(self, cache):
```

Comment:
This is confusing, why would I want to pop packages that are upgradable from a dict origin -> package? Maybe the class needs a better name that explains the purpose better